public/js/pimcore/helpers.js

```diff
@@ -876,7 +876,21 @@ pimcore.helpers.download = function (url) {
         pimcore.settings.showCloseConfirmation = true;
     }, 1000);
 
-    location.href = url;
+    let iframe = document.getElementById('download_helper_iframe');
+    if (!iframe) {
+        iframe = document.createElement('iframe');
+        iframe.setAttribute('id', 'download_helper_iframe');
+        document.body.appendChild(iframe);
+    }
+    iframe.src = url;
+
+    document.getElementById('download_helper_iframe').onload = function() {
+        if(iframe.src !== 'about:blank') {
+            const title = iframe.contentDocument.title;
+            pimcore.helpers.showNotification(t('error'), title, 'error');
+            iframe.src = 'about:blank';
+        }
+    }
```

Comment:
```suggestion
    iframe.onload = function() {
          const title = iframe.contentDocument.title;
          pimcore.helpers.showNotification(t('error'), title, 'error');
    }
```
this may doing the trick in a simpler way.
When the download is triggered, there's no DOM `onload` event at all

Reply:
let's also add a js comment on the purpose of this `iframe`